From b7a7df111b7f5630545b246f49c2502c7adffeab Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Wed, 27 Aug 2014 20:55:18 -0400
Subject: [usb] xhci: fix incorrect type in assignment in handle_device_notification()

Message-id: <20140827205518.3990.179.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 89286
O-Subject: [PATCH RHEL7.1 BZ1123119 1/2] xhci: fix incorrect type in assignment in handle_device_notification()
Bugzilla: 1123119
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1123119

Description:
------------
This patch converts Event TRB's 3rd field, which has type le32, to CPU
byteorder before using it to retrieve the Slot ID with TRB_TO_SLOT_ID macro.
This bug was found using sparse.

Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>

RHEL Version Found:
-------------------
7.0

Upstream Status:
----------------
http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=7e76ad431545d013911ddc744843118b43d01e89

----------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-ring.c b/drivers/usb/host/xhci-ring.c
index 55fc0c3..e187943 100644
--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -1611,7 +1611,7 @@ static void handle_device_notification(struct xhci_hcd *xhci,
  u32 slot_id;
  struct usb_device *udev;
 
- slot_id = TRB_TO_SLOT_ID(event->generic.field[3]);
+ slot_id = TRB_TO_SLOT_ID(le32_to_cpu(event->generic.field[3]));
  if (!xhci->devs[slot_id]) {
   xhci_warn(xhci, "Device Notification event for "
     "unused slot %u\n", slot_id);
-- 
1.7.1